src/cargo/util/config.rs

```diff
@@ -51,6 +53,9 @@ impl Config {
             extra_verbose: Cell::new(false),
             frozen: Cell::new(false),
             locked: Cell::new(false),
+            // This should be called early on in the process, so in theory the
+            // unsafety is ok here. (taken ownership of random fds)
+            jobserver: unsafe { jobserver::Client::from_env() },
```

Comment:
There should be at most one job server, and it's not entirely obvious that we call `Config::new` exactly once. Perhaps, we can wrap jobserver creation in a `jobserver_singleton` function, which uses `ONCE_INIT` or `lazy_static!` internally?